### DISCUSSION TITLE
Read_gml  error

### DISCUSSION THREAD

**Discussion by daibitao19** [networkx/networkx#5096] [Q&A]
Thanks for your attention. I met some trouble when I read gml file. 

The specific errpr is on display. Can you teach me how to read this gml file successfully? This gml file given by my friend and 
she told me that it's exported by networkx.
Thanks a lot

---------------------------------------------------------------------------
NetworkXError                             Traceback (most recent call last)
<ipython-input-22-b435dd4fa3ee> in <module>
----> 1 G=nx.read_gml('gml/GzEco9WCd.gml')

<decorator-gen-801> in read_gml(path, label, destringizer)

~\Anaconda3\lib\site-packages\networkx\utils\decorators.py in _open_file(func_to_be_decorated, *args, **kwargs)
    237         # Finally, we call the original function, making sure to close the fobj
    238         try:
--> 239             result = func_to_be_decorated(*new_args, **kwargs)
    240         finally:
    241             if close_fobj:

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in read_gml(path, label, destringizer)
    179             yield line
    180 
--> 181     G = parse_gml_lines(filter_lines(path), label, destringizer)
    182     return G
    183 

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in parse_gml_lines(lines, label, destringizer)
    409 
    410     tokens = tokenize()
--> 411     graph = parse_graph()
    412 
    413     directed = graph.pop("directed", False)

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in parse_graph()
    398 
    399     def parse_graph():
--> 400         curr_token, dct = parse_kv(next(tokens))
    401         if curr_token.category is not None:  # EOF
    402             unexpected(curr_token, "EOF")

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in parse_kv(curr_token)
    353                 curr_token = next(tokens)
    354             elif category == Pattern.DICT_START:
--> 355                 curr_token, value = parse_dict(curr_token)
    356             else:
    357                 # Allow for string convertible id and label values

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in parse_dict(curr_token)
    392         curr_token = consume(curr_token, Pattern.DICT_START, "'['")
    393         # dict contents
--> 394         curr_token, dct = parse_kv(curr_token)
    395         # dict end
    396         curr_token = consume(curr_token, Pattern.DICT_END, "']'")

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in parse_kv(curr_token)
    353                 curr_token = next(tokens)
    354             elif category == Pattern.DICT_START:
--> 355                 curr_token, value = parse_dict(curr_token)
    356             else:
    357                 # Allow for string convertible id and label values

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in parse_dict(curr_token)
    394         curr_token, dct = parse_kv(curr_token)
    395         # dict end
--> 396         curr_token = consume(curr_token, Pattern.DICT_END, "']'")
    397         return curr_token, dct
    398 

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in consume(curr_token, category, expected)
    333         if curr_token.category == category:
    334             return next(tokens)
--> 335         unexpected(curr_token, expected)
    336 
    337     def parse_kv(curr_token):

~\Anaconda3\lib\site-packages\networkx\readwrite\gml.py in unexpected(curr_token, expected)
    328         category, value, lineno, pos = curr_token
    329         value = repr(value) if value is not None else "EOF"
--> 330         raise NetworkXError(f"expected {expected}, found {value} at ({lineno}, {pos})")
    331 
    332     def consume(curr_token, category, expected):

NetworkXError: expected ']', found EOF at (7, 1)

**Comment by rossbar** [comment]
It's not possible to say what's wrong specifically without looking at the file itself, but the exception message is very detailed and should give you a good idea where to look. The exception indicates that it unexpectedly ran into the end of the file on line 7. This means that the data is likely malformed at or prior to the 7th line in the file. I'd start looking there.

**Comment by MridulS** [comment]
> This gml file given by my friend and she told me that it's exported by networkx.

Also it's possible that the export was done with a different version of networkx. Ask them to also share the networkx version and try to use that to read in the GML file. You can find the networkx version with:
```python
import networkx as nx
print(nx.__version__)
```
